<launch> 
  <arg name="mav1"                  default="sam"/>
  <arg name="mav2"                  default="bucky"/> 
  
  <arg name="map1_topic"            default="/$(arg mav1)/rtabmap/mapData"/>
  <arg name="map2_topic"            default="/$(arg mav2)/rtabmap/mapData"/>
  <!--<arg name="map2_topic"            default="none"/>-->
  <arg name="map3_topic"            default="none"/>

  <arg name="base_frame"            default="world"/>
  <arg name="db_location"           default="/home/jacob/.ros/mapmerge.db"/>
  <!--<arg name="config_path"         default=""/>-->
  <arg name="map_merge_frequency"   default="5"/>
  <arg name="odom_linear_variance"  default="0.0001"/>
  <arg name="odom_angular_variance" default="0.0005"/>

  <node name="map_merger" pkg="map_merger" type="map_merger" output="screen">
    <param name="map1_topic"            value="$(arg map1_topic)"/>
    <param name="map2_topic"            value="$(arg map2_topic)"/>
    <param name="map3_topic"            value="$(arg map3_topic)"/>
    <param name="base_frame"            value="$(arg base_frame)"/>
    <param name="db_location"           value="$(arg db_location)"/>
    <param name="map_merge_frequency"   value="$(arg map_merge_frequency)"/>
    <param name="odom_linear_variance"  value="$(arg odom_linear_variance)"/>
    <param name="odom_angular_variance" value="$(arg odom_angular_variance)"/>
  </node>
  
  <!-- Run RVIZ -->
  <node type="rviz" name="rviz" pkg="rviz" 
    args="-d $(find map_merger)/params/map_merger.rviz" />
</launch>
